### DISCUSSION TITLE
suggest to differentiate between ENTER and RETURN keypress

### DISCUSSION THREAD

**Discussion by vacarogit** [sumatrapdfreader/sumatrapdf#3837] [Ideas]
Need Sumatra PDF to treat Enter(numpad area) and Return as differentkeystroke e.g.  assignment
Smplayer did that well!

Thanks for consideration!

Shortcuts [
	[
		Cmd = CmdExit
		Key = Enter
	]
]

v.s.
Shortcuts [
	[
		Cmd = CmdToggleFullscreen
		Key = Enter
	]
]

**Comment by kjk** [comment]
Tell me which virtual key it is and I'll add it: https://learn.microsoft.com/en-us/windows/win32/inputdev/virtual-key-codes

However, I don't see anything in that list and I don't have keyboard with numpad to test.

**Reply by GitHubRulesOK** [reply]
Interesting issue both keys return 0D (13) return=enter however Javascript can tell the difference by looking at the "location"
so main enter key = 0:13 but Numpad = 3:13 see https://www.toptal.com/developers/keycode

```
{
 "key": "Enter",
 "keyCode": 13,
 "which": 13,
 "code": "Enter",
 "location": 0,
 "altKey": false,
 "ctrlKey": false,
 "metaKey": false,
 "shiftKey": false,
 "repeat": false
}
```

```
{
 "key": "Enter",
 "keyCode": 13,
 "which": 13,
 "code": "NumpadEnter",
 "location": 3,
 "altKey": false,
 "ctrlKey": false,
 "metaKey": false,
 "shiftKey": false,
 "repeat": false
}
```
